test/support/otlp_endpoint_test.rb

```diff
@@ -0,0 +1,228 @@
+      assert_entity_endpoint_nil
+      assert_headers_nil
+    end
+
+    it 'correct formatted SW_APM_SERVICE_KEY with headers set' do
+      ENV['SW_APM_SERVICE_KEY'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234:my_service'
+      ENV['OTEL_EXPORTER_OTLP_METRICS_HEADERS'] = 'authorization=Bearer 0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+      ENV['OTEL_EXPORTER_OTLP_HEADERS'] = 'authorization=Bearer 0123456789abcde0123456789abcde0123456789abcde0123456789abcde1236'
+
+      token_setup
+
+      _(@endpoint.instance_variable_get(:@token)).must_equal '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_METRICS_HEADERS', nil)).must_equal 'authorization=Bearer 0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_LOGS_HEADERS', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_TRACES_HEADERS', nil))
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil)).must_equal 'authorization=Bearer 0123456789abcde0123456789abcde0123456789abcde0123456789abcde1236'
+
+      assert_entity_endpoint_nil
+    end
+  end
+
+  describe 'config_endpoint' do
+    def endpoint_setup
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      matches = ENV['SW_APM_COLLECTOR'].to_s.match(SolarWindsAPM::OTLPEndPoint::SWO_APM_ENDPOINT_REGEX)
+      endpoint.resolve_get_setting_endpoint(matches)
+      %w[TRACES METRICS LOGS].each { |data_type| endpoint.configure_otlp_endpoint(data_type, matches) }
+    end
+
+    it 'no OTEL ENDPOINT and no SW_APM_COLLECTOR' do
+      endpoint_setup
+
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil))
+
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443/v1/traces'
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_METRICS_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443/v1/metrics'
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_LOGS_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443/v1/logs'
+      _(ENV.fetch('SW_APM_COLLECTOR', nil)).must_equal 'apm.collector.na-01.cloud.solarwinds.com:443'
+    end
+
+    it 'OTEL ENDPOINT to local and no SW_APM_COLLECTOR' do
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'http://localhost:4317'
+
+      endpoint_setup
+
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_METRICS_ENDPOINT', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_LOGS_ENDPOINT', nil))
+
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil)).must_equal 'http://localhost:4317'
+      _(ENV.fetch('SW_APM_COLLECTOR', nil)).must_equal 'apm.collector.na-01.cloud.solarwinds.com:443'
+    end
+
+    it 'OTEL ENDPOINT to otel and with SW_APM_COLLECTOR' do
+      ENV['SW_APM_COLLECTOR'] = 'apm.collector.na-02.cloud.solarwinds.com:443'
```

Comment:
we should also have tests to check if the end user has explicitly set an SW APM collector endpoint that does NOT have the port, things still work.